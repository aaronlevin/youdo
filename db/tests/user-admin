\i create.sql

\echo create
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', 'yddb-tests/user-admin')
returning id
    \gset txn
insert into yd_userV (txnid, obj)
values (:txnid, row('Alice'))
returning id
    \gset user
end;
select * from yd_userV;
select * from yd_user;

\echo update
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', 'yddb-tests/user-admin')
returning id
    \gset txn
insert into yd_userV (txnid, id, obj)
values (:txnid, :userid, row('Bob'));
end;
select * from yd_userV;
select * from yd_user;

\echo delete
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', 'yddb-tests/user-admin')
returning id
    \gset txn
insert into yd_userV (txnid, id, obj)
values (:txnid, :userid, null);
end;
select * from yd_userV;
select * from yd_user;
