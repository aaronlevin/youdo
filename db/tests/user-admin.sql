\i create.sql

\echo create
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', :'script');
insert into yd_user (name) values ('Alice')
returning id
    \gset user
end;
select * from yd_user_v;
select * from yd_user;

\echo update
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', :'script');
update yd_user set name = 'Bob' where id = :userid;
end;
select * from yd_user_v;
select * from yd_user;

\echo delete
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', :'script');
delete from yd_user where id = :userid;
end;
select * from yd_user_v;
select * from yd_user;
