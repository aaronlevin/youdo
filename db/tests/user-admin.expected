create
yddb|1|0|f
Alice|2|1|f
yddb|1|0
Alice|2|1
update
yddb|1|0|f
Alice|2|1|f
Bob|3|1|f
yddb|1|0
Bob|3|1
delete
yddb|1|0|f
Alice|2|1|f
Bob|3|1|f
Bob|4|1|t
yddb|1|0
